lib/modules/manager/pixi/extract.ts

```diff
@@ -0,0 +1,69 @@
+import { logger } from '../../../logger';
+import { getSiblingFileName, localPathExists } from '../../../util/fs';
+import { Result } from '../../../util/result';
+import { Toml } from '../../../util/schema-utils';
+import { PyProjectSchema } from '../pep621/schema';
+import type { PackageFileContent } from '../types';
+import { type PixiConfig, PixiToml } from './schema';
+
+const PyProjectToml = Toml.pipe(PyProjectSchema);
+
+function getUserPixiConfig(
+  content: string,
+  packageFile: string,
+): null | PixiConfig {
+  if (packageFile.endsWith('pyproject.toml')) {
+    const { val, err } = Result.parse(content, PyProjectToml).unwrap();
+    if (err) {
+      logger.debug({ packageFile, err }, `error parsing ${packageFile}`);
+      return null;
+    }
+
+    return val.tool?.pixi ?? null;
+  }
+
+  if (packageFile.endsWith('pixi.toml')) {
+    const { val, err } = Result.parse(content, PixiToml).unwrap();
+    if (err) {
+      logger.debug({ packageFile, err }, `error parsing ${packageFile}`);
+      return null;
+    }
+
+    return val;
+  }
+
+  const { val: maybePyprojectConfig } = Result.parse(
+    content,
+    PyProjectToml,
+  ).unwrap();
+
+  if (maybePyprojectConfig?.tool?.pixi) {
+    return maybePyprojectConfig.tool.pixi;
+  }
+
+  const { val: maybePixiConfig } = Result.parse(content, PixiToml).unwrap();
+  return maybePixiConfig ?? null;
+}
+
+export async function extractPackageFile(
+  content: string,
+  packageFile: string,
+): Promise<PackageFileContent | null> {
+  logger.trace(`pixi.extractPackageFile(${packageFile})`);
+
+  const config = getUserPixiConfig(content, packageFile);
+  if (!config) {
+    return null;
+  }
+
+  const lockfileName = getSiblingFileName(packageFile, 'pixi.lock');
+  const lockFiles: string[] = [];
+  if (await localPathExists(lockfileName)) {
+    lockFiles.push(lockfileName);
+  }
+
+  return {
+    lockFiles,
+    deps: [],
+  };
```

Comment:
```suggestion
  if (!await localPathExists(lockfileName)) {
    return null;
  }

  return {
    lockFiles: [lockfileName],
    deps: [],
  };
```
return null if no deps and no lockfile

Reply:
we will return deps in the futhre, and it will cause unnecessary diff then

Reply:
also it make us hard to test if it's unsupport toml or can't find anything.